### PR TITLE
ensure the token randomness length in all cases

### PR DESCRIPTION
**Pull request by sasurobert**
ensure the token randomness length in all cases

### CODE REVIEW

**Review by dorin-iancu — DISMISSED** [review]

File: vm/systemSmartContracts/esdt.go [hunk 1]
```diff
@@ -409,16 +409,20 @@ func isTokenNameHumanReadable(tokenName []byte) bool {
...
 	newRandomAsBigInt := big.NewInt(0).SetBytes(newRandomForTicker)
 
 	one := big.NewInt(1)
 	for i := 0; i < numOfRetriesForIdentifier; i++ {
-		encoded := hex.EncodeToString(newRandomAsBigInt.Bytes())
-		newIdentifier := append(tickerPrefix, []byte(encoded)...)
+		encoded := []byte(hex.EncodeToString(newRandomAsBigInt.Bytes()))
+		lenEncoded := len(encoded)
+		if lenEncoded < encodedRandomSequenceLength {
+			encoded = append(make([]byte, encodedRandomSequenceLength-lenEncoded), encoded...)
```
Comment (andrei-marinica) [inline vm/systemSmartContracts/esdt.go:423]: wouldn't it make more sense to prepend the character '0' instead of \x00 bytes?

Comment (raduchis) [reply vm/systemSmartContracts/esdt.go:423]: :+1:

**Review by raduchis — APPROVED** [review]

File: vm/systemSmartContracts/esdt.go (hunk 1 above)
Comment (andrei-marinica) [reply vm/systemSmartContracts/esdt.go:423]: `fmt.Sprintf("%06x", big.NewInt(123))` does it in one line, the only problem is I don't think you can provide the number of digits as a variable. But I think encodedRandomSequenceLength is a constant anyway, up to you to decide

File: vm/systemSmartContracts/esdt.go [hunk 2]
```diff
@@ -409,16 +409,20 @@ func isTokenNameHumanReadable(tokenName []byte) bool {
 
 func (e *esdt) createNewTokenIdentifier(caller []byte, ticker []byte) ([]byte, error) {
 	newRandomBase := append(caller, e.eei.BlockChainHook().CurrentRandomSeed()...)
-	newRandom := sha256.Sha256{}.Compute(string(newRandomBase))
+	newRandom := e.hasher.Compute(string(newRandomBase))
```
Comment (iulianpascalau) [inline vm/systemSmartContracts/esdt.go:412]: is this backwards compatible?

File: vm/systemSmartContracts/esdt.go (hunk 1 above)
Comment (iulianpascalau) [reply vm/systemSmartContracts/esdt.go:423]: ...or you could use bytes.Repeat function

File: vm/systemSmartContracts/esdt.go (hunk 2 above)
Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go:412]: with mainnet it is.

File: vm/systemSmartContracts/esdt.go (hunk 1 above)
Comment (sasurobert) [reply vm/systemSmartContracts/esdt.go]: done as @andrei-marinica mentioned.

**Review by raduchis — APPROVED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Review by iulianpascalau — APPROVED** [review]

**Review by bogdan-rosianu — APPROVED** [review]
System tests passed

**Review by gabi-vuls — APPROVED** [review]
System test passed

**Pull request merged**